### PR TITLE
Test files now  pytest

### PR DESCRIPTION
**Pull request by PZES**
**Description**

Changed test files to use pytest

**Verification**

Ran pytest on the files

### CODE REVIEW

File: selfdrive/car/ford/tests/test_ford.py [hunk 1]
```diff
@@ -37,45 +40,109 @@
...
-class TestFordFW(unittest.TestCase):
+class TestFordFW:
   def test_fw_query_config(self):
     for (ecu, addr, subaddr) in FW_QUERY_CONFIG.extra_ecus:
-      self.assertIn(ecu, ECU_ADDRESSES, "Unknown ECU")
-      self.assertEqual(addr, ECU_ADDRESSES[ecu], "ECU address mismatch")
-      self.assertIsNone(subaddr, "Unexpected ECU subaddress")
+      assert ecu in ECU_ADDRESSES, "Unknown ECU"
+      assert addr == ECU_ADDRESSES[ecu], "ECU address mismatch"
+      assert subaddr, "Unexpected ECU subaddress" is None
```
Comment (sshane) [inline selfdrive/car/ford/tests/test_ford.py:51]:
```suggestion
      assert subaddr is None, "Unexpected ECU subaddress"
```

File: selfdrive/car/ford/tests/test_ford.py [hunk 2]
```diff
@@ -1,12 +1,15 @@
 #!/usr/bin/env python3
```
Comment (sshane) [inline selfdrive/car/ford/tests/test_ford.py:1]: you can rebase/merge master into your branch so you don't have these duplicate diffs on your PR

File: selfdrive/car/ford/tests/test_ford.py [hunk 3]
```diff
@@ -37,45 +40,109 @@
...
-      # Software part number takes the form: PREFIX-CORE-SUFFIX
-      # Prefix changes based on the family of part. It includes the model year
-      #   and likely the platform.
-      # Core identifies the type of the item (e.g. 14D003 = PSCM, 14C204 = PCM).
-      # Suffix specifies the version of the part. -AA would be followed by -AB.
-      #   Small increments in the suffix are usually compatible.
-      # Details: https://forscan.org/forum/viewtopic.php?p=70008#p70008
+      assert ecu in ECU_PART_NUMBER, "Unexpected ECU"
+      assert addr == ECU_ADDRESSES[ecu], "ECU address mismatch"
+      assert subaddr, "Unexpected ECU subaddress" is None
```
Comment (sshane) [inline selfdrive/car/ford/tests/test_ford.py:58]: same

**Comment by sshane** [comment]
I recall we had an issue with pytest.skip earlier, can you make sure that's no longer an issue where you replaced unittest.skip? Perhaps that was because we just ran with unittest

**Comment by adeebshihadeh** [comment]
Looks like there's still a bunch of unittest references:
```
batman:twopilot$ rg unittest selfdrive/
selfdrive/updated/tests/test_base.py
9:import unittest
10:from unittest import mock
58:      raise unittest.SkipTest

selfdrive/thermald/tests/test_fan_controller.py
2:from unittest.mock import Mock, patch

selfdrive/thermald/tests/test_power_monitoring.py
2:from unittest.mock import patch

selfdrive/athena/tests/test_registration.py
5:from unittest import mock

selfdrive/athena/tests/test_athenad_ping.py
7:from unittest import mock

selfdrive/athena/tests/test_athenad.py
16:from unittest import mock

selfdrive/car/tests/test_models.py
7:import unittest
158:      raise unittest.SkipTest
163:        raise unittest.SkipTest

selfdrive/car/tests/test_docs.py
6:import unittest
87:          raise unittest.SkipTest

selfdrive/car/tests/test_lateral_limits.py
6:import unittest
38:      raise unittest.SkipTest("Platform is behind dashcamOnly")
42:      raise unittest.SkipTest
45:      raise unittest.SkipTest
78:  result = unittest.main(exit=False)

selfdrive/car/tests/test_fw_fingerprint.py
5:import unittest
8:from unittest import mock
66:      raise unittest.SkipTest("Brand does not implement custom fuzzy fingerprinting function")
88:      raise unittest.SkipTest("Car model has no compatible ECUs for fuzzy matching")

selfdrive/boardd/tests/test_pandad.py
5:import unittest
113:      raise unittest.SkipTest("SPI test")

selfdrive/car/hyundai/tests/test_hyundai.py
4:import unittest
72:          raise unittest.SkipTest("Skip checking Santa Cruz for its parts")
158:            raise unittest.SkipTest("No part numbers for car model")

selfdrive/car/ford/tests/test_ford.py
3:import unittest
149:  unittest.main()

selfdrive/car/volkswagen/tests/test_volkswagen.py
4:import unittest
66:  unittest.main()

selfdrive/ui/tests/test_ui/run.py
11:import unittest
191:  unittest.main(exit=False)
```

File: selfdrive/car/ford/tests/test_ford.py (hunk 1 above)
Comment (sshane) [reply selfdrive/car/ford/tests/test_ford.py:51]: think adeeb's merge reverted your fix

**Comment by sshane** [comment]
I suggest we add `"unittest".msg = "use pytest!"` to ruff's `banned-api`

**Comment by adeebshihadeh** [comment]
Closing, this is a very simple change that shouldn't require as much review as this has gotten so far.

**Pull request closed without merging**